008_team_6_agent_design/008_team_6_agent_design.tex

```diff
@@ -1 +1,1021 @@
+    \label{fig:socialMotives}
+\end{figure}
+
+
+\subsection{Changing Social Motives}
+Placing an agent into one of these fixed categories for the entire duration of the game would be limiting and unrealistic. This led to the addition of the first layer of complexity: given that an agent may be initially assigned a social motive, it should be plausible for the agent to change their social motive based on their experience. This concept brings forth the interesting duality of ``nature vs nurture'' \cite{natNur}. One could also think about this in terms of ``genotype vs phenotype''. For example, an agent may `naturally' or `genetically' be born a collectivist but may be incentivised to change its values and act selfishly if it becomes desperate due to a lack of food. In this environment, the factors that influence an agent’s social motive are its current HP and current floor, as both reflect an implicit lack of food.
+
+The process of changing an agent’s social motive based on certain factors can be represented as a weighted sum of these factors. In this case, where the related parameters are the current HP value and the current floor, the weighted sum can be achieved by defining a parameters vector \textit{p} and a weights vector \textit{w}. The weighted sum is then given by the product $w^\top{}p$. These are shown below:
+
+
+\begin{equation}
+    \begin{gathered}
+    p = [\mathit{currentHP, currentFloor}],\quad w=[\mathit{weightHP, weightFloor}] \\ 
+    w^\top{}p = \mathit{weightHP*currentHP + weightFloor*currentFloor}
+    \end{gathered}
+\end{equation}
+
+
+
+\subsubsection{Feature Transformations} \label{sec:featTran}
+
+Instead of using the values of \textit{currentHP} and \textit{currentFloor} directly, we use transformation functions to map them into an output domain between 0 and 1. These transformations additionally allow us to associate a non-linear importance to a given factor. We refer to the resulting factors as \textit{hpScore} and \textit{floorScore} respectively.
+
+The $hpScore$ is calculated as
+
+\begin{equation}
+    \mathit{hpScore} = 1 - \frac{\mathit{currentHP}}{\mathit{maxHP}}
+\end{equation}
+
+yielding a simple linear mapping (\textit{HP} $\in$ [0,100] $\rightarrow$ [1,0]), where a higher \textit{currentHP} maps to a lower \textit{hpScore}, resulting in a tendency for the agent to behave more altruistically.
+
+Calculating the \textit{floorScore} requires knowledge of the maximum floor in the tower. As this information is not available to the agent, it will use its memory of the past floors it has been on to forecast this value. Furthermore, the mapping is non-linear, with lower floors (higher number) being `punished' more; how much more is determined by the parameter $\lambda$ where a greater $\lambda$ weights lower floors disproportionately more. This gives:
```

Comment:
```suggestion
Calculating the \textit{floorScore} requires knowledge of the maximum floor in the tower. As this information is not available to the agent, it will use its memory of the past floors it has been on to forecast this value. Furthermore, the mapping is non-linear, with lower floors (higher number) being `punished' more; how much more is determined by the parameter $\lambda$, where a greater $\lambda$ weights lower floors disproportionately more. This gives:
```